File: app/src/main/java/com/lazycouple/restapiclient/deligate/MyApplication.java
package com.lazycouple.restapiclient.deligate;

import android.app.Application;

import com.amuyu.logger.DefaultLogPrinter;
import com.amuyu.logger.Logger;
import com.lazycouple.restapiclient.Injection;

import io.realm.Realm;
import io.realm.RealmConfiguration;


/**
 * Created by noco on 2016-10-12.
 */
public class MyApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        Logger.addLogPrinter(new DefaultLogPrinter(this));
        initRealm();
    }

    public void initRealm() {
        Realm.init(this);
        RealmConfiguration realmConfig = Injection.provideRealmConfiguration(this);
        Realm.setDefaultConfiguration(realmConfig);
        Realm realm;
        try {
            realm = Realm.getDefaultInstance();
        } catch (IllegalArgumentException iae) {
            Logger.w(iae);
            Realm.deleteRealm(realmConfig);
            realm = Realm.getDefaultInstance();
        }
        realm.close();
    }
}

File: app/src/main/kotlin/com/lazycouple/restapiclient/deligate/MyApplication.kt
package com.lazycouple.restapiclient.deligate

import android.app.Application
import com.amuyu.logger.DefaultLogPrinter
import com.amuyu.logger.Logger
import com.facebook.stetho.Stetho
import com.lazycouple.restapiclient.Injection
import com.lazycouple.restapiclient.ui.ApplicationComponent
import com.lazycouple.restapiclient.ui.ApplicationModule
import com.lazycouple.restapiclient.ui.DaggerApplicationComponent
import io.realm.Realm


/**
 * Created by noco on 2016-10-12.
 */
class MyApplication : Application() {

    lateinit var appComponent: ApplicationComponent
        private set

    override fun onCreate() {
        super.onCreate()

        appComponent = DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(this))
                .build()
        app = this

        Logger.addLogPrinter(DefaultLogPrinter(this))
        initRealm()
        Stetho.initializeWithDefaults(this)
    }

    fun initRealm() {
        Realm.init(this)
        val realmConfig = Injection.provideRealmConfiguration(this)
        Realm.setDefaultConfiguration(realmConfig)
        var realm: Realm
        try {
            realm = Realm.getDefaultInstance()
        } catch (iae: IllegalArgumentException) {
            Logger.w(iae)
            Realm.deleteRealm(realmConfig)
            realm = Realm.getDefaultInstance()
        }

        realm.close()
    }

    companion object {
        lateinit var app: MyApplication
            private set
    }
}
